src/main/java/com/server/nodak/domain/post/repository/PostRepositoryImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.server.nodak.domain.post.repository;
+
+import static com.server.nodak.domain.post.domain.QPost.post;
+import static com.server.nodak.domain.vote.domain.QVoteHistory.voteHistory;
+
+import com.querydsl.core.BooleanBuilder;
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.jpa.JPAExpressions;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import com.server.nodak.domain.post.dto.PostSearchRequest;
+import com.server.nodak.domain.post.dto.PostSearchResponse;
+import com.server.nodak.domain.post.dto.QPostSearchResponse;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+
+@RequiredArgsConstructor
+@Slf4j
+public class PostRepositoryImpl implements PostRepositoryCustom {
```

Comment:
`PostRepositoryImpl` 은 `PostRepository` 구현 클래스인 것처럼 느껴져요.
이름을 명확히 하는 것도 좋을 것 같습니다!

```suggestion
public class PostRepositoryCustomImpl implements PostRepositoryCustom {
```

Reply:
변경했으면 좋은 네이밍 컨벤션도 함께 말씀해주시면 좋을 것 같습니다

Reply:
제가 잘못 이해했군요🤣 그렇다면 그대로 가도 될것 같습니다!